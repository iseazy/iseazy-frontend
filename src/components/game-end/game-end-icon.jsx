export const GameEndIcon = () => (
    <svg xmlns="http://www.w3.org/2000/svg" width={32} height={32} >
        <defs>
        <style>
            {
            ".game-end__icon{fill:none;stroke:#999;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px}"
            }
        </style>
        </defs>
        <g transform="translate(-1 -1)">
        <circle
            className="game-end__icon"
            cx={15}
            cy={15}
            r={15}
            transform="translate(2 2)"
        />
        <path className="game-end__icon" d="M17 8v9l6 3" />
        </g>
    </svg>
)