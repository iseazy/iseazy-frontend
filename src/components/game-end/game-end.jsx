import { getFormattedTime } from '../../utils';
import { GameEndIcon } from './game-end-icon';

export const GameEnd = ({ time, onClick }) => (
    <div className="game-end">
            <div className="game-end__popup">
                <div className="game-end__title">
                    <div>Â¡Completado!</div>
                    <GameEndIcon />
                    <h1>{ getFormattedTime(time) }</h1>
                </div>
            <button className="game-end__button" onClick={ onClick}>Jugar otra vez</button>
        </div>
    </div>
);
