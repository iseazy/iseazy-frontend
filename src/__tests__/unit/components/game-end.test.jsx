import { render, fireEvent, screen } from '@testing-library/react';
import { GameEnd } from '../../../components/game-end/game-end';

jest.mock('../../../components/game-end/game-end-icon', () => ({
    GameEndIcon: () => 'mockedIcon'
}));

describe('<GameEnd />', () => {
    const testProps = {
        time: 150,
        onClick: () => {}
    };
    
    it('should render component', () => {
        const { asFragment } = render(<GameEnd { ...testProps }/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should be clicked', () => {
        const onClick = jest.fn();
        render(<GameEnd { ...testProps } onClick={ onClick }/>);

        fireEvent.click(screen.getByRole('button'));

        expect(onClick).toHaveBeenCalled();
    });
});
