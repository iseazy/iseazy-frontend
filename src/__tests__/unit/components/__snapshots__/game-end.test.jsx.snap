// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameEnd /> should render component 1`] = `
<DocumentFragment>
  <div
    class="game-end"
  >
    <div
      class="game-end__popup"
    >
      <div
        class="game-end__title"
      >
        <div>
          Â¡Completado!
        </div>
        mockedIcon
        <h1>
          2:30
        </h1>
      </div>
      <button
        class="game-end__button"
      >
        Jugar otra vez
      </button>
    </div>
  </div>
</DocumentFragment>
`;
